specs/core/0_beacon-chain.md

```diff
@@ -946,27 +978,32 @@ def get_crosslink_committees_at_slot(state: BeaconState,
     """
     Returns the list of ``(committee, shard)`` tuples for the ``slot``.
     """
-    state_epoch_slot = state.slot - (state.slot % EPOCH_LENGTH)
-    assert state_epoch_slot <= slot + EPOCH_LENGTH
-    assert slot < state_epoch_slot + EPOCH_LENGTH
-    offset = slot % EPOCH_LENGTH
-
-    if slot < state_epoch_slot:
-        committees_per_slot = get_previous_epoch_committee_count_per_slot(state)
-        shuffling = get_shuffling(
-            state.previous_epoch_seed,
-            state.validator_registry,
-            state.previous_epoch_calculation_slot,
-        )
-        slot_start_shard = (state.previous_epoch_start_shard + committees_per_slot * offset) % SHARD_COUNT
+    epoch = slot_to_epoch(slot)
+    current_epoch = get_current_epoch(state)
+    previous_epoch = current_epoch - 1 if epoch > GENESIS_EPOCH else current_epoch
```

Comment:
It seems the "previous" is relative to "current". Should we use `current_epoch` instead of `epoch` in this line?
```suggestion
    previous_epoch = current_epoch - 1 if current_epoch > GENESIS_EPOCH else current_epoch
```